code/core/src/manager/components/sidebar/Search.stories.tsx

```diff
@@ -23,77 +22,101 @@ const getLastViewed = () =>
+      >
+        <LayoutProvider>
+          <IconSymbols />
+          {storyFn()}
+        </LayoutProvider>
+      </ManagerContext.Provider>
     ),
   ],
 } satisfies Meta<typeof Search>;
+
 export default meta;
 
-const baseProps = {
-  dataset,
-  clearLastViewed: action('clear'),
-  getLastViewed: () => [] as Selection[],
+type Story = StoryObj<typeof meta>;
+
+export const Simple: Story = {};
+
+export const SimpleWithCreateButton: Story = {
+  args: {
+    showCreateStoryButton: true,
+  },
+};
+
+export const FilledIn: Story = {
+  args: {
+    initialQuery: 'Search query',
+  },
+};
+
+export const LastViewed: Story = {
+  args: {
+    getLastViewed,
+  },
+};
+
+export const ShortcutsDisabled: Story = {
+  args: {
+    enableShortcuts: false,
+  },
+};
+
+export const Searching: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = await within(canvasElement);
+    const search = await canvas.findByPlaceholderText('Find components');
+    await userEvent.clear(search);
+    await userEvent.type(search, 'foo');
+    expect(setQueryParams).toHaveBeenCalledWith({
+      search: 'foo',
+    });
+  },
 };
 
-export const Simple: StoryFn = () => <Search {...baseProps}>{() => null}</Search>;
-
-export const SimpleWithCreateButton: StoryFn = () => (
-  <Search {...baseProps} showCreateStoryButton={true}>
-    {() => null}
-  </Search>
-);
-
-export const FilledIn: StoryFn = () => (
-  <Search {...baseProps} initialQuery="Search query">
-    {() => <SearchResults {...noResults} />}
-  </Search>
-);
-
-export const LastViewed: StoryFn = () => (
-  <Search {...baseProps} getLastViewed={getLastViewed}>
-    {({ query, results, closeMenu, getMenuProps, getItemProps, highlightedIndex }) => (
-      <SearchResults
-        query={query}
-        results={results}
-        closeMenu={closeMenu}
-        getMenuProps={getMenuProps}
-        getItemProps={getItemProps}
-        highlightedIndex={highlightedIndex}
-      />
-    )}
-  </Search>
-);
-
-export const ShortcutsDisabled: StoryFn = () => (
-  <Search {...baseProps} enableShortcuts={false}>
-    {() => null}
-  </Search>
-);
-
-export const CustomShortcuts: StoryFn = () => <Search {...baseProps}>{() => null}</Search>;
-
-CustomShortcuts.decorators = [
-  (storyFn) => (
-    <ManagerContext.Provider
-      value={
-        {
-          api: {
-            getShortcutKeys: () => ({ search: ['control', 'shift', 's'] }),
-          } as API,
-        } as any
-      }
-    >
-      {storyFn()}
-    </ManagerContext.Provider>
-  ),
-];
+export const Clearing: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = await within(canvasElement);
+    const search = await canvas.findByPlaceholderText('Find components');
+    await userEvent.clear(search);
+    await userEvent.type(search, 'foo');
+
+    const clearIcon = await canvas.findByTitle('Clear search');
+    await userEvent.click(clearIcon);
+
+    expect(setQueryParams).toHaveBeenCalledWith({ search: null });
```

Comment:
Based on https://github.com/storybookjs/storybook/pull/16527/files#diff-39f5915dc511a2d421234f378dc2a5efbaef09acfe73d81f95e440838c1326f1R286 this should be `undefined` instead of `null`, which is also evident by the failing test: https://www.chromatic.com/test?appId=635781f3500dd2c49e189caf&id=66c22bd5c11b883b85898ff9